src/main/java/sg/ncl/MainController.java

```diff
@@ -1191,6 +1191,7 @@ public String teamProfile(@PathVariable String teamId, Model model, HttpSession
         model.addAttribute("team", team);
         model.addAttribute("owner", team.getOwner());
         model.addAttribute("membersList", team.getMembersList());
+        session.setAttribute("originalTeam", team);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make "Team2" serializable or don't store it in the session. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2441)